types/utils.go

```diff
@@ -154,3 +155,45 @@ func ContainsString(s []string, e string) bool {
 	}
 	return false
 }
+
+func CompareVersionStrings(verStr1, verStr2 string) (int, error) {
+	ver1 := strings.Split(verStr1, ".")
+	ver2 := strings.Split(verStr2, ".")
+	lenVer1 := len(ver1)
+	lenVer2 := len(ver2)
+
+	numChunks := lenVer1
```

Comment:
can we just do a math.Min operation?

Reply:
`math.Min` is for float64.  I think golang does not have a built-in min function for integers.